map/bookmark_helpers.cpp

```diff
@@ -245,17 +245,18 @@ std::string RemoveInvalidSymbols(std::string const & name)
   return strings::ToUtf8(filtered);
 }
 
-std::string GenerateUniqueFileName(const std::string & path, std::string name, std::string const & ext)
+std::string GenerateUniqueFileName(std::string const & path, std::string name, std::string_view const ext)
 {
   // Remove extension, if file name already contains it.
   if (strings::EndsWith(name, ext))
     name.resize(name.size() - ext.size());
 
   size_t counter = 1;
   std::string suffix;
-  while (Platform::IsFileExistsByFullPath(base::JoinPath(path, name + suffix + ext)))
+
+  while (Platform::IsFileExistsByFullPath(base::JoinPath(path, name.append(suffix).append(ext))))
```

Comment:
There is a bug here. We modify name var, but should not.

Looks like we are making very controversial changes for some very _shadow_ profit ..

Reply:
Yep, this issue was caught by unit-test, I fixed it.